src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -324,8 +324,15 @@ public CSharpOperationFactory(SemanticModel semanticModel)
                     };
                     return new NoneOperation(children, _semanticModel, boundNode.Syntax, type: type, constantValue, isImplicit: isImplicit);
                 case BoundKind.ValuePlaceholder:
+                    // The only supported use of BoundValuePlaceholder is within a collection expression as we use it
+                    // to represents the elements passed to the collection builder creation methods.  We can hit these
+                    // creation methods when producing the .CreationArguments for the ICollectionExpressionOperation.
+                    // Note: the caller will end up stripping this off when producing the CreationArguments, so it will
+                    // not actually leak to the user.  But this ends up keeping the logic simple between that callsite
+                    // and this code which actually hits all the arguments passed along.
+                    Debug.Assert(boundNode.Syntax is CollectionExpressionSyntax);
```

Comment:
Thanks for this comment. I think I now see that I was not quite getting a nuance here, and no longer think we need to add a placeholder kind, since we never expose it.